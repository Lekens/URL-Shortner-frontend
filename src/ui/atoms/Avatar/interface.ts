export interface IAvatar {
  src: string;
  size: number;
  extraClassName?: string;
}
