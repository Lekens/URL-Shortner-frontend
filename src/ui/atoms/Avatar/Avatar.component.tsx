import { useMemo } from 'react';

import cn from 'classnames';

import { IAvatar } from 'ui/atoms/Avatar/interface';

import styles from './styles.module.scss';

function Avatar({ src, size, extraClassName }: IAvatar) {
  const avatarStyle = useMemo(
    () => cn(styles.wrapper, extraClassName),
    [extraClassName],
  );

  return (
    <div className={avatarStyle} style={{ width: size, height: size }}>
      {src && <img src={src} alt={src} className={styles.img} />}
    </div>
  );
}
export default Avatar;
