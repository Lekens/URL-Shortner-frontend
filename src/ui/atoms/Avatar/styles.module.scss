@import 'features/common/assets/styles/variables.module.scss';

.wrapper {
  background-color: $dark-blue;
  border-image-width: 0;
  border-radius: 50%;
  .img {
    width: 100%;
    height: 100%;
    border-radius: inherit;
    object-fit: cover;
  }
}
