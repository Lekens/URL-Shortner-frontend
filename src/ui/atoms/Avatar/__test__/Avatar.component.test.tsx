import React from 'react';

import { shallow } from 'enzyme';

import { render } from '@testing-library/react';
import Avatar from 'ui/atoms/Avatar/Avatar.component';

describe('Avatar Component', () => {
  it('It should match snapshot with image src = https://s.dou.ua/img/avatars/200x200_158710.jpg', () => {
    const data = {
      src: 'https://s.dou.ua/img/avatars/200x200_158710.jpg',
      size: 30,
      extraClassName: 'HelloClass',
    };
    const { container } = render(<Avatar {...data} />);

    expect(container).toMatchSnapshot();

    const imageWrapper = shallow(<Avatar {...data} />);
    expect(imageWrapper.find('img').props().src).toBe(
      'https://s.dou.ua/img/avatars/200x200_158710.jpg',
    );
    expect(imageWrapper.find('img').props().alt).toBe(
      'https://s.dou.ua/img/avatars/200x200_158710.jpg',
    );
  });
});
