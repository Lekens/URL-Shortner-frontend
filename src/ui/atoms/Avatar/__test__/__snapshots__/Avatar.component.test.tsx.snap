// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Component It should match snapshot with image src = https://s.dou.ua/img/avatars/200x200_158710.jpg 1`] = `
<div>
  <div
    class="wrapper HelloClass"
    style="width: 30px; height: 30px;"
  >
    <img
      alt="https://s.dou.ua/img/avatars/200x200_158710.jpg"
      class="img"
      src="https://s.dou.ua/img/avatars/200x200_158710.jpg"
    />
  </div>
</div>
`;
